### PR TITLE
PCHR-1690: Fix loading spinner in job roles tab

### PR DESCRIPTION
**Pull request by bmnepali**
This PR contains the fixes:
1. Loading spinner in job roles tab forever when no contract is available
2. Remove setting messages code on adding, updating and deleting job role 
3. Display loader(spinner) as in job Contract tab.

### Issue:
While fetching contract for the given contact, availability of number of contracts  was not checked which was resulting the display of spinner continuously.

### Solution:
1. While fetching contract for the given contact, availability of no. of contracts shall be checked.
2. If there exists one or more contract -> modified to display the spinner
3. If there are not contract for the given contact -> modified to stop displaying the spinner
4. Adding loader directive and display loader/spinner as in hr job contract tab
5. Reusing the loader/spinner form reqangular module.

### Screenshot:  
1. If no contract is available, doesn't displays the continuous spinner in job roles tab.
np-spinner

2. No message displayed on adding new job role
add

3. No message displayed on  updating  a job role
edit

4. No message displayed on deleting a job role
delete

5. Add loader(spinner) to display as in job contract tab
loader

### CODE REVIEW

File: com.civicrm.hrjobroles/js/src/job-roles/app.js [hunk 1]
```diff
@@ -10,13 +10,14 @@ define([
...
     'job-roles/filters/filters',
+    'job-roles/directives/loader',
     'job-roles/services/services',
     'job-roles/controllers/hr-job-roles-controller',
     'job-roles/controllers/modal-dialog',
     'job-roles/services/hr-job-roles-service',
     'job-roles/services/date-validation',
     'job-roles/services/hr-job-roles-service-filter',
-    'job-roles/directives/example',
+    'job-roles/directives/example'
```
Comment (AkA84) [inline com.civicrm.hrjobroles/js/src/job-roles/app.js]: What's this for?

File: com.civicrm.hrjobroles/js/src/job-roles/directives/loader.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
+define([
+  'job-roles/directives/directives'
+], function(directives) {
+  'use strict';
+
+  directives.directive('jobrolesLoader', ['$rootScope', '$log', function($rootScope, $log) {
```
Comment (AkA84) [inline com.civicrm.hrjobroles/js/src/job-roles/directives/loader.js]: This is not good. You have basically copy-pasted the same directive that we have on job contract. It's highly unmaintainable and contributes to the bloating of the code base.

There is an extension, [org.civicrm.reqangular](https://github.com/civicrm/civihr/tree/staging/org.civicrm.reqangular) whose job is to be a repo of all the common modules (directives, filters, etc) that might be shared between extension. We already have a [`loading`](https://github.com/civicrm/civihr/blob/staging/org.civicrm.reqangular/src/common/directives/loading.js) directive that displays a spinner. See if you can use that. If for whatever reason you cannot, then you need to see why and to integrate `loading` with this `jobrolesLoader`

Comment (bmnepali) [inline com.civicrm.hrjobroles/js/src/job-roles/directives/loader.js]: Got it, will do!

**Comment by mi-kon** [comment]
@bmnepali i think it's alright the way it is now. Later we will need to design how Job roles and job contracts tabs look when there are no contracts or roles

**Pull request merged**